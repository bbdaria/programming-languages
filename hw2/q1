fun w1 f g x =  g (f x);
fun w2 x f = f(f x);
fun w3 f a b = f(a,b);
fun w4 a b = a-b<0.01;
fun w5 f lst = map (fn innerList => map f innerList) lst;
fun w6 a b c = if a= true then c else b;
fun w7 () = 5;
fun w8 f g a d = g(f a);
fun w9 lst = length(lst);
fun w10 f [] = [] | w10 f (x::xs) = (f x) :: (w10 f xs);
