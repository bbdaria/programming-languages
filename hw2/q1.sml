fun sig1 f g x =  g (f x);
fun sig2 x f = f(f x);
fun sig3 f a b = f(a,b);
fun sig4 a b = a - b < 0.01;
fun sig5 f lst = map (fn innerList => map f innerList) lst;
fun sig6 a b c = if a = true then c else b;
fun sig7 () = 5;
fun sig8 f g a d = g(f a);
fun sig9 lst = length(lst);
fun sig10 f [] = [] | sig10 f (x::xs) = (f x) :: (sig10 f xs);
