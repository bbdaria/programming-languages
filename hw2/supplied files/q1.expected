val it = () : unit
val it = fn : ('a -> 'b) -> ('b -> 'c) -> 'a -> 'c
val it = fn : 'a -> ('a -> 'a) -> 'a
val it = fn : ('a * 'b -> 'c) -> 'a -> 'b -> 'c
val it = fn : real -> real -> bool
val it = fn : ('a -> 'b) -> 'a list list -> 'b list list
val it = fn : bool -> 'a -> 'a -> 'a
val it = fn : unit -> int
val it = fn : ('a -> 'b) -> ('b -> 'c) -> 'a -> 'd -> 'c
val it = fn : 'a list -> int
val it = fn : ('a -> 'b) -> 'a list -> 'b list
- 
